src/main/java/com/teragrep/aer_02/EventDataConsumer.java

```diff
@@ -0,0 +1,253 @@
+import com.teragrep.aer_02.config.source.Sourceable;
+import com.teragrep.rlo_14.Facility;
+import com.teragrep.rlo_14.SDElement;
+import com.teragrep.rlo_14.Severity;
+import com.teragrep.rlo_14.SyslogMessage;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.dropwizard.DropwizardExports;
+import io.prometheus.client.exporter.MetricsServlet;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiConsumer;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+final class EventDataConsumer implements AutoCloseable, BiConsumer<EventData, PartitionContext> {
+
+    // Note: Checkpointing is handled automatically.
+    private final Output output;
+    private final String realHostName;
+    private final SyslogConfig syslogConfig;
+    private final MetricRegistry metricRegistry;
+    private final JmxReporter jmxReporter;
+    private final Slf4jReporter slf4jReporter;
+    private final Server jettyServer;
+
+    // metrics
+    private final AtomicLong records = new AtomicLong();
+    private final AtomicLong allSize = new AtomicLong();
+
+    EventDataConsumer(Sourceable configSource, int prometheusPort) {
+        this(configSource, new MetricRegistry(), prometheusPort);
+    }
+
+    EventDataConsumer(Sourceable configSource, MetricRegistry metricRegistry, int prometheusPort) {
+        this(
+                configSource,
+                new DefaultOutput("defaultOutput", new RelpConfig(configSource), metricRegistry),
+                metricRegistry,
+                prometheusPort
+        );
+    }
+
+    EventDataConsumer(Sourceable configSource, Output output, MetricRegistry metricRegistry, int prometheusPort) {
+        this.metricRegistry = metricRegistry;
+        this.output = output;
+        this.realHostName = getRealHostName();
+        this.syslogConfig = new SyslogConfig(configSource);
+
+        this.jmxReporter = JmxReporter.forRegistry(metricRegistry).build();
+        this.slf4jReporter = Slf4jReporter
+                .forRegistry(metricRegistry)
+                .outputTo(LoggerFactory.getLogger(EventDataConsumer.class))
+                .convertRatesTo(TimeUnit.SECONDS)
+                .convertDurationsTo(TimeUnit.MILLISECONDS)
+                .build();
+        jettyServer = new Server(prometheusPort);
+        startMetrics();
+
+        metricRegistry
+                .register(name(EventDataConsumer.class, "estimated-data-depth"), (Gauge<Double>) () -> (allSize.get() / records.doubleValue()) / records.doubleValue());
```

Comment:
bogus calculation, please remove or justify the logic behind the calculation

Reply:
these were removed in new pr